sound/soc/sof/ipc4-pcm.c

```diff
@@ -140,27 +142,27 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 		goto free;
 
 	/* else set the final state in the DSP */
-	ret = sof_ipc4_set_multi_pipeline_state(sdev, state, data);
+	ret = sof_ipc4_set_multi_pipeline_state(sdev, state, trigger_list);
 	if (ret < 0) {
 		dev_err(sdev->dev, "failed to set final state %d for all pipelines\n", state);
 		goto free;
 	}
 
 	/* update final state for all pipelines that were just triggered */
-	for (i = 0; i < data->count; i++) {
+	for (i = 0; i < trigger_list->count; i++) {
 		for (j = 0; j < pipeline_list->count; j++) {
 			spipe = pipeline_list->pipelines[j];
 			pipe_widget = spipe->pipe_widget;
 
-			if (data->pipeline_ids[i] == pipe_widget->instance_id) {
+			if (trigger_list->pipeline_ids[i] == pipe_widget->instance_id) {
 				pipeline->state = state;
```

Comment:
This contradicts the comment on front of the outermost for(): we are updating *the* pipeline state and do not iterate through all of them, I think this is a bug in the original code and warrant  a fixup patch first?

Reply:
a `pipeline = pipe_widget->private;` is missing at line 152/154.

Reply:
> This contradicts the comment on front of the outermost for(): we are updating _the_ pipeline state and do not iterate through all of them, I think this is a bug in the original code and warrant a fixup patch first?

im sorry not getting the bug part. We are iterating through both lists to only set the final state for the pipeline IDs in trigger_lsit here right?